pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -320,6 +320,17 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, r.Update(context.TODO(), cd)
 	}
 
+	// Set platform label on the ClusterDeployment
+	platform := getClusterPlatform(cd)
+	if cd.Labels == nil {
```

Comment:
From a code organization standpoint, I would rather see this inside the next if statement. In other words, only make the map if the platform label does not match the cd platform (understanding that if the labels field is nil that they won't match anyway).